cartographer/io/fake_file_writer_test.cc

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <vector>
+
+#include "cartographer/io/fake_file_writer.h"
+#include "gtest/gtest.h"
+namespace cartographer {
+namespace io {
+namespace {
+
+TEST(FakeFileWriter, CloseStream) {
+  auto on_close_output = std::make_shared<std::string>();
+  FakeFileWriter writer("file", on_close_output);
+  EXPECT_EQ(writer.GetFilename(), "file");
+  EXPECT_TRUE(writer.Close());
+  EXPECT_EQ(*on_close_output, "");
+  EXPECT_FALSE(writer.Close());
+}
+
+TEST(FakeFileWriter, WriteHeader) {
+  auto on_close_output = std::make_shared<std::string>();
+  const std::string header("dummy header");
```

Comment:
You could add a `\0` in the middle of the string here for example.

Reply:
I actually can not use std::string("dummy\0 header") to construct the test cases, as the std::string internally will only store "dummy" in this case. That's also why I added test cases using std::vector<char> as an input.